### ISSUE TITLE
upload-release when logged out fails poorly

### ISSUE THREAD

**Issue by benmoss** [cloudfoundry/bosh-cli#140]
if you are not logged into your bosh director and try to run `bosh upload-release` you end up getting this error:

```
Using environment 'https://192.168.11.2:25555' as anonymous user
########################################################## 100.00% 64.36 MB/s 1s

Uploading release file:
  Performing request POST 'https://192.168.11.2:25555/releases?':
    Performing POST request:
      Seeking to begining of seekable request body during attempt 0: seek /Users/pivotal/.bosh/tmp/bosh-platform-disk-TarballCompressor-CompressSpecificFilesInDir657716434: bad file descriptor

Exit code 1
```

this seems like the wrong error, since i'm pretty sure its just an authorization issue and not a bad file descriptor. if i run `bosh releases` i get this:

```
bosh releases
Using environment 'https://192.168.11.2:25555' as anonymous user

Finding releases:
  Director responded with non-successful status code '401' response 'Not authorized: '/releases'
'

Exit code 1
```

which is more helpful.

**Comment by dpb587** [comment]
I've run into this a couple times and been very confused by the error message as well.

**Comment by benmoss** [comment]
```
$ bosh -v
version 2.0.5-ec671ee-2017-02-18T01:11:19Z
```

**Comment by dpb587-pivotal** [comment]
Related: #91

**Comment by dpb587-pivotal** [comment]
Story: https://www.pivotaltracker.com/story/show/143258597

**Comment by anEXPer** [comment]
Is that link broken?

**Comment by dpb587-pivotal** [comment]
Link is broken; I'm not sure which story it was referring to. There haven't been changes around this code, and I'm not sure of a suggested code fix.

**Comment by benmoss** [comment]
We just ran into this very terrible error when trying to deploy while logged out:

```
Using environment '192.168.50.6' as client 'admin'

Using deployment 'credhub'

Fetching diff result:
  Performing request POST 'https://192.168.50.6:25555/deployments/credhub/diff?redact=true':
    Performing POST request:
      Retry: Post https://192.168.50.6:25555/deployments/credhub/diff?redact=true: http: ContentLength=10280 with Body length 0

Exit code 1
```

**Comment by anEXPer** [comment]
So, can we please get a story for this? Much as I enjoy the slow burn of more people running into it and coming to this issue and then finding out that the story is gone.

**Comment by cppforlife** [comment]
updated story: https://www.pivotaltracker.com/story/show/156744770

**Comment by hiremaga** [comment]
We just saw a very similar issue when we were correctly logged in to the bosh director. Talking to @cppforlife it sounds like this error will also be displayed when using a bosh client without the [bosh.releases.upload](http://bosh.io/docs/director-users-uaa-perms/#release-uploader) scope.

**Comment by mfine30** [comment]
Looks like the error messaged that is returned now mentioned it is an authorization issue rather than the misleading error before. This should be fixed in v5.3.1 of the CLI.

```
$ bosh upload-release /tmp/bosh-vsphere-cpi.tgz
Using environment 'https://10.0.0.6:25555' as client 'admin'

                                                                           0.00%
Uploading release file:
  Performing request POST 'https://10.0.0.6:25555/releases?':
    Performing POST request:
      Requesting token via client credentials grant: UAA responded with non-successful status code '401' response '{"error":"unauthorized","error_description":"Bad credentials"}'

Exit code 1
```
